lib/utils/combineJSON.js

```diff
@@ -60,14 +60,19 @@ function combineJSON(arr, deep, collision, source, parsers=[]) {
 
       // Iterate over custom parsers, if the file path matches the parser's
       // pattern regex, use it's parse function to generate the object
-      parsers.forEach(({pattern, parse}) => {
-        if (resolvedPath.match(pattern)) {
-          file_content = parse({
+      // Allows async parsers
+      for (let i = 0; i < parsers.length; i++) {
```

Comment:
I think 
```
for(const parser of parsers.filter(p=>resolvePath.match(p.pattern))){
  file_content = await parser.parse({
    contents: fs.readFileSync(resolvedPath, {encoding:'UTF-8'}),
    filePath: resolvedPath
  });
}
```
 is more readable IMHO

Reply:
`for of` I agree, one-lining the conditional into it, debatable, but I'll take it :)